lib/react.dart

```diff
@@ -371,6 +375,312 @@ abstract class Component {
   dynamic render();
 }
 
+abstract class Component2Adapter {
+  void setState(dynamic newState, SetStateCallback callback);
+  void forceUpdate(SetStateCallback callback);
+}
+
+/// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/react-component.html)
+/// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/react-component.html#reference)
+abstract class Component2<T extends Map> implements Component {
```

Comment:
Why do we implement Component?

Reply:
For compatibility purposes with existing APIs that accept/return a `Component`. It's not ideal to have to stub in all these unused methods and things, but I think it's worth it for this class to be compatible.